### PR TITLE
[FIX] (website_)event_booth_sale: fix e-commerce displayed price

### PR DESCRIPTION
**Pull request by pko-odoo**
PURPOSE

Since this change (https://github.com/odoo/odoo/pull/75843), the displayed price
on the e-commerce was the product price and not the sum of the prices of all the
selected booths. Pricelist weren't taken into account either.
This commit fix these issues.

LINKS

Task-2657480

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_event_booth_sale/models/sale_order.py [hunk 1]
```diff
@@ -23,33 +23,47 @@ def _cart_find_product_line(self, product_id=None, line_id=None, **kwargs):
         return lines
 
     def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
+        if self._context.get('pricelist') != self.pricelist_id.id:
+            self = self.with_context(pricelist=self.pricelist_id.id)
```
Comment (awa-odoo) [inline addons/website_event_booth_sale/models/sale_order.py:27]: Hmmm.

That introduces context management that will go up to the super method.
Is that intended? Seems a bit louche.

File: addons/event_booth_sale/models/sale_order_line.py [hunk 2]
```diff
@@ -92,11 +92,6 @@ def _get_sale_order_line_multiline_description_sale(self):
...
     def _get_display_price(self):
         if self.event_booth_pending_ids and self.event_id:
-            company = self.event_id.company_id or self.env.company
-            currency = company.currency_id
-            total_price = sum([booth.price for booth in self.event_booth_pending_ids])
-            return currency._convert(
-                total_price, self.order_id.currency_id,
-                self.order_id.company_id or self.env.company.id,
-                self.order_id.date_order or fields.Date.today())
+            return sum(booth.price_reduce for booth in
```
Comment (awa-odoo) [inline addons/event_booth_sale/models/sale_order_line.py:95]: I think a test case would be most welcome if possible ^^

File: addons/website_event_booth_sale/models/sale_order.py [hunk 3]
```diff
@@ -23,33 +23,47 @@ def _cart_find_product_line(self, product_id=None, line_id=None, **kwargs):
...
-            new_registration_ids = [Command.create({
-                                        'event_booth_id': booth.id,
-                                        **kwargs.get('registration_values'),
-                                    }) for booth in booths]
-            if order_line:
-                event_booth_registration_ids = [Command.delete(reg.id)
-                                                for reg in order_line.event_booth_registration_ids] + new_registration_ids
-            else:
-                event_booth_registration_ids = new_registration_ids
+            if order_line.event_booth_pending_ids.ids != event_booth_pending_ids:
```
Comment (awa-odoo) [inline addons/website_event_booth_sale/models/sale_order.py:36]: This change looks a bit unrelated to the price display.
Can you explain?

Would we create registrations too many times before?

File: addons/website_event_booth_sale/models/sale_order.py [hunk 4]
```diff
@@ -23,33 +23,47 @@ def _cart_find_product_line(self, product_id=None, line_id=None, **kwargs):
...
 
         return values
 
     def _cart_update(self, product_id=None, line_id=None, add_qty=0, set_qty=0, **kwargs):
         values = {}
+        if line_id:
+            line = self.env['sale.order.line'].browse(line_id)
+            booths = line.event_booth_pending_ids
+            if booths:
+                kwargs.update({'event_booth_pending_ids': booths.ids})
```
Comment (awa-odoo) [inline addons/website_event_booth_sale/models/sale_order.py:66]: Updating the kwargs manually before calling super is a bit strange.
Also feels strange that we need to store that information into kwargs as it's already present on the so.line that is passed further down the call chain anyway.

**Review by awa-odoo — COMMENTED** [review]
Hello @pko-odoo 

Validation as promised.
Sorry for the delay.

Cheers!

File: addons/website_event_booth_sale/models/sale_order.py (hunk 3 above)
Comment (pko-odoo) [reply addons/website_event_booth_sale/models/sale_order.py:33]: Indeed it's not related to the price issue !
I just consider useless to delete and recreate the registrations each time the method is called

**Review by pko-odoo — COMMENTED** [review]
Hello @awa-odoo 

I answered with a review to avoid spam ^^
I dug a little more deeply in my code since it has been a few months to try to understand what I did back then...

File: addons/website_event_booth_sale/models/sale_order.py (hunk 4 above)
Comment (awa-odoo) [reply addons/website_event_booth_sale/models/sale_order.py:63]: You mean in the "update_pricelist" use case?
I see that in that case it also passes the product and "add_qty=0".

So maybe instead we could add it there:
```py
if product.detailed_type == 'event_booth' and set_qty > 1:
    set_qty = 1
    values['warning'] = _('You cannot manually change the quantity of an Event Booth product.')
elif product.detailed_type == 'event_booth' and line_id and add_qty == 0 and not kwargs.get('event_booth_pending_ids'):
    # when updating the pricelist, the website_sale module call this "_cart_update" method without passing the event.booth we need to update in the kwargs
    # -> we manually set the argument to make sure the price is updated in "_website_product_id_change"
    kwargs['event_booth_pending_ids'] = self.env['sale.order.line'].browse(line_id).event_booth_pending_ids.ids
```

Maybe that is not correct though, you would need to double-check (and a unit test would be ideal of course ^^).

File: addons/website_event_booth_sale/models/sale_order.py [hunk 5]
```diff
@@ -29,27 +29,38 @@ def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
...
+                                        }) for booth in booths]
+                if order_line:
+                    event_booth_registration_ids = [Command.delete(reg.id)
+                                                    for reg in order_line.event_booth_registration_ids] + new_registration_ids
+                else:
+                    event_booth_registration_ids = new_registration_ids
+                values.update(event_booth_registration_ids=event_booth_registration_ids)
 
+            if self.pricelist_id.discount_policy == 'without_discount':
+                price_unit = sum(booth.price for booth in booths)
```
Comment (awa-odoo) [inline addons/website_event_booth_sale/models/sale_order.py:46]: By the way, I would assume that "price_unit" is the price for a single unit, is that intended to sum for every booth here?
Maybe I'm misunderstanding that.

**Comment by C3POdoo** [comment]
Dear @pko-odoo,

Thank you for your contribution but we are closing it due to inactivity.

We apology if we could not look at your request in time.
If the contribution still makes sense, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._
